test/e2e-go/features/catchup/basicCatchup_test.go

```diff
@@ -331,3 +331,92 @@ func shutdownClonedNode(nodeDataDir string, f *fixtures.RestClientFixture, t *te
 		os.RemoveAll(nodeDataDir)
 	}
 }
+
+// TestBasicCatchupCompletes confirms the the catchup eventually completes and stops.
+func TestBasicCatchupCompletes(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	// Make the network progress faster
+	consensus := make(config.ConsensusProtocols)
+	fastProtocol := config.Consensus[protocol.ConsensusCurrentVersion]
+	fastProtocol.ApprovedUpgrades = map[protocol.ConsensusVersion]uint64{}
+	fastProtocol.AgreementFilterTimeoutPeriod0 = 400 * time.Millisecond
+	fastProtocol.AgreementFilterTimeout = 400 * time.Millisecond
+	consensus[protocol.ConsensusCurrentVersion] = fastProtocol
+
+	// Setup the fixture with the modified fast consensus
+	var fixture fixtures.RestClientFixture
+	fixture.SetConsensus(consensus)
+	fixture.Setup(t, filepath.Join("nettemplates", "TwoNodes100Second.json"))
+	defer fixture.Shutdown()
+
+	// Get 2nd node so we wait until we know they're at target block
+	nc, err := fixture.GetNodeController("Node")
+	a.NoError(err)
+
+	// Let the network make some progress.
+	// Make it long enough so the catchup to it is longer than a single round agreement
+	a.NoError(err)
+	waitForRound := uint64(100)
+
+	// Now prepare a third node
+	cloneDataDir := filepath.Join(fixture.PrimaryDataDir(), "../clone")
+	cloneLedger := false
+	err = fixture.NC.Clone(cloneDataDir, cloneLedger)
+	a.NoError(err)
+
+	// Wait for the network to make some progess.
+	err = fixture.ClientWaitForRoundWithTimeout(fixture.GetAlgodClientForController(nc), waitForRound)
+	a.NoError(err)
+
+	// Start the third node to catchup.
+	startTime := time.Now()
+	cloneClient, err := fixture.StartNode(cloneDataDir)
+	a.NoError(err)
+	defer shutdownClonedNode(cloneDataDir, &fixture, t)
+
+	// Wait for it to catchup
+	err = fixture.LibGoalFixture.ClientWaitForRoundWithTimeout(cloneClient, waitForRound)
```

Comment:
this isn't accurate enough; you want to wait until the round number from both nodes is the same.
i.e. while you're waiting for the node to catchup, the network is keep making progress. you want to make sure that the round number aligns.